astropy/io/ascii/cds.py

```diff
@@ -119,6 +119,10 @@ def get_cols(self, lines):
                 unit = match.group('units')
                 if unit == '---':
                     col.unit = None  # "---" is the marker for no unit in CDS table
+                elif unit[0] == '[' and unit[-1] == ']':
```

Comment:
Is this comparison going to catch other non-CDS stuff and cause unintended side effects?

Reply:
No because this function is only run when the program guesses that the provided format is CDS.